### DISCUSSION TITLE
IsHidden method in class Filament\Actions\Concerns\CanBeHidden, whether isAuthorized should be called first.

### DISCUSSION THREAD

**Discussion by wxfjamdc** [filamentphp/filament#8261] [General]
public function isHidden(): bool
    {
        if ($this->evaluate($this->isHidden)) {
            return true;
        }

        if (! $this->evaluate($this->isVisible)) {
            return true;
        }

        return ! $this->isAuthorized();
    }

isAuthorized is judged at the end, and some buttons, such as DeleteAction, have been configured with hidden in the setUp method, which will cause the permission judgment to fail.

**Comment by danharrin** [comment]
I don't understand, please expand. They are separate checks that should not impact each other. We safely ensure that if any check fails, the user cannot access it.

**Reply by wxfjamdc** [reply]
My fault, I thought wrong

**Reply by wxfjamdc** [reply]
Please delete this discussion, there is no need to exist anymore
